privaterelay/types.py

```diff
@@ -1,5 +1,59 @@
 """Types for the privaterelay app"""
 
-from typing import Literal
+from typing import Literal, TypedDict
 
 RELAY_CHANNEL_NAME = Literal["local", "dev", "stage", "prod"]
+
+# django-csp 4.0: types for CONTENT_SECURITY_POLICY in settings.py
+
+# See https://github.com/mozilla/django-csp/blob/main/csp/utils.py
+CSP_DIRECTIVES_T = TypedDict(
+    "CSP_DIRECTIVES_T",
+    {
+        # Fetch Directives
+        "child-src": list[str],
+        "connect-src": list[str],
+        "default-src": list[str],
+        "script-src": list[str],
+        "script-src-attr": list[str],
+        "script-src-elem": list[str],
+        "object-src": list[str],
+        "style-src": list[str],
+        "style-src-attr": list[str],
+        "style-src-elem": list[str],
+        "font-src": list[str],
+        "frame-src": list[str],
+        "img-src": list[str],
+        "manifest-src": list[str],
+        "media-src": list[str],
+        "prefetch-src": list[str],  # Deprecated.
+        # Document Directives
+        "base-uri": list[str],
+        "plugin-types": list[str],  # Deprecated.
+        "sandbox": list[str],
+        # Navigation Directives
+        "form-action": list[str],
+        "frame-ancestors": list[str],
+        "navigate-to": list[str],
+        # Reporting Directives
+        "report-uri": str,
+        "report-to": list[str],
+        "require-sri-for": list[str],
+        # Trusted Types Directives
+        "require-trusted-types-for": list[str],
+        "trusted-types": list[str],
+        # Other Directives
+        "webrtc": list[str],
+        "worker-src": list[str],
+        # Directives Defined in Other Documents
+        "upgrade-insecure-requests": list[str],
+        "block-all-mixed-content": list[str],  # Deprecated.
```

Comment:
These last 2 are actually booleans.

Reply:
I'm going to review the spec, I think some of the other types are wrong as well.

@robhudson If you want me to add typing annotations upstream (https://github.com/mozilla/django-csp/issues/198), this could be the seed. 28 files should take me about a day to annotate.